concrete-core-fixture/src/fixture/lwe_ciphertext_vector_conversion.rs

```diff
@@ -0,0 +1,181 @@
+                },
+                LweCiphertextVectorConversionParameters {
+                    noise: Variance(0.00000001),
+                    lwe_dimension: LweDimension(300),
+                    lwe_ciphertext_count: LweCiphertextCount(100),
+                },
+                LweCiphertextVectorConversionParameters {
+                    noise: Variance(0.00000001),
+                    lwe_dimension: LweDimension(600),
+                    lwe_ciphertext_count: LweCiphertextCount(100),
+                },
+                LweCiphertextVectorConversionParameters {
+                    noise: Variance(0.00000001),
+                    lwe_dimension: LweDimension(1000),
+                    lwe_ciphertext_count: LweCiphertextCount(100),
+                },
+                LweCiphertextVectorConversionParameters {
+                    noise: Variance(0.00000001),
+                    lwe_dimension: LweDimension(3000),
+                    lwe_ciphertext_count: LweCiphertextCount(100),
+                },
+                LweCiphertextVectorConversionParameters {
+                    noise: Variance(0.00000001),
+                    lwe_dimension: LweDimension(6000),
+                    lwe_ciphertext_count: LweCiphertextCount(100),
+                },
+            ]
+            .into_iter(),
+        )
+    }
+
+    fn generate_random_repetition_prototypes(
+        parameters: &Self::Parameters,
+        maker: &mut Maker,
+    ) -> Self::RepetitionPrototypes {
+        let proto_secret_key = maker.new_lwe_secret_key(parameters.lwe_dimension);
+        (proto_secret_key,)
+    }
+
+    fn generate_random_sample_prototypes(
+        parameters: &Self::Parameters,
+        maker: &mut Maker,
+        repetition_proto: &Self::RepetitionPrototypes,
+    ) -> Self::SamplePrototypes {
+        let (key,) = repetition_proto;
+        let raw_plaintext_vector = Precision::Raw::uniform_vec(parameters.lwe_ciphertext_count.0);
+        let proto_plaintext_vector =
+            maker.transform_raw_vec_to_plaintext_vector(raw_plaintext_vector.as_slice());
+        let proto_ciphertext_vector = maker.encrypt_plaintext_vector_to_lwe_ciphertext_vector(
+            &key,
+            &proto_plaintext_vector,
+            parameters.noise,
+        );
+        (proto_ciphertext_vector,)
+    }
+
+    fn prepare_context(
+        _parameters: &Self::Parameters,
+        maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+        sample_proto: &Self::SamplePrototypes,
+    ) -> Self::PreExecutionContext {
+        let (proto_ciphertext_vector,) = sample_proto;
+        (maker.synthesize_lwe_ciphertext_vector(proto_ciphertext_vector),)
+    }
+
+    fn execute_engine(
+        _parameters: &Self::Parameters,
+        _engine: &mut Engine,
+        context: Self::PreExecutionContext,
+    ) -> Self::PostExecutionContext {
+        let (ciphertext_vector,) = context;
+        (ciphertext_vector,)
```

Comment:
I don't understand. Is this somehow implicitly calling the engine? And why are both Pre- and PostExecutionContext of the type `CudaCiphertextVector`? I would have expected some conversion happening here.

Reply:
Yes I was a bit stuck here, because actually the Synthesis stage handles the conversions. So I don't have an engine to execute anymore... I don't know how Alex wanted to do this.

Reply:
So what I did is that I left the conversions in the synthesis but I simply don't execute any engine. I can open an issue to check that with him when he's back.

Reply:
Maybe it makes sense to ask Alex. I think to do this properly we would need either two makers (one for each ciphertext type) or a maker that can deal with both somehow. Not sure what the best way is, but this seems like it's abusing the framework. Should I approve this then and we open an issue for this?

Reply:
Yes, I'd rather do that so that the tutorials work.